### ISSUE TITLE
Layout issues when working with Angular 5 + Polymer

### ISSUE THREAD

**Issue by mutantkeyboard** [VizArtJS/vizart-path#1]
Hi @BigFatDog . 
I've started using your components in Angular 5 app that I'm building at the moment. 
I'm currently experiencing some weird issues. 
When I set up margins like this: 
![2](https://user-images.githubusercontent.com/1167494/35150321-49048ec8-fd1a-11e7-9444-13beceec4929.png)

then Chrome returns an error:

![1](https://user-images.githubusercontent.com/1167494/35150336-587ecc7e-fd1a-11e7-9d1a-70af3984752c.png)

Code I use for parsing the data is very simple: 
![3](https://user-images.githubusercontent.com/1167494/35150364-6a7010d2-fd1a-11e7-8bb4-4537e5b24798.png)

If I change the margins to negative values for top, and bottom like this: 
![4](https://user-images.githubusercontent.com/1167494/35150390-7ee7cafa-fd1a-11e7-9cd6-e13db31f17a0.png)

then I see the coordinates, but not ticks, nor brushes.

![5](https://user-images.githubusercontent.com/1167494/35150434-aa488a0e-fd1a-11e7-8456-ed15b589dee3.png)

Finally. If I modify the HTML attribute by adding some styling to it like this: 
![6](https://user-images.githubusercontent.com/1167494/35150427-a5728656-fd1a-11e7-9510-6b03c0d52ef9.png)

Then my end result is like this: 
![7](https://user-images.githubusercontent.com/1167494/35150454-b4214e3a-fd1a-11e7-8105-f5e6626b572a.png)

I've never experienced something like this, so any insight is very welcome.

**Comment by BigFatDog** [comment]
Hello @mutantkeyboard,

The root causes might be:
1. css position issue
2. Not good width and height are set.

My recommendation is try to set a fix width and height for the chart firstly (a quick change for test purpose only). If it works, the logic of rendering is good. 

You can set width and height for charts in two ways:
1. set on chart div directly using css.

```javascript
this._chart = new ParallelCoordinates('#chart')
<div id='chart' style='width=600px, height=400px'>
```

2. Set in chart config

```javascript
const config = {
    margin: ....
    width: 800,
    height: 600 // in numbers, not 600px
}
this._chart = new ParallelCoordinates('#chart', config)
```

My assumptions are based on my following observations on the errors that you provided:
1. Error #1 The error in chrome: 
```javascript
<rect> height attribute: ...
```

The overall chart height is insufficient, it causes all brushes (in forms of SVG rects) height calculated in negative values. Therefore the root case is width&height

2. Error #2

It could be height

3. Error #3

Ticks and lines are apart. It could be highly possible the css positioning conflict. However, we'd better try out the width/height solution before resolving this.

Hope that works. If error persists and it is an specific error with Angular 5, please let me know. I'll try to reproduce it by setting up an Angular 5 app myself.

**Comment by mutantkeyboard** [comment]
Hi @BigFatDog 

I've scaffolded a new Angular 5 application, and used Bootstrap 4 only. (No Material + Polymer)

I still have a similar problem, though it says that I have a NULL value, and nullValueSeparator is not set, which apparently isn't true, since I set it up in configuration to 'bottom'
![trouble](https://user-images.githubusercontent.com/1167494/35227640-e026af5e-ff8e-11e7-9584-1eb61b320c72.png)

![trouble2](https://user-images.githubusercontent.com/1167494/35227685-ffdb9102-ff8e-11e7-8e3b-f01d601e4f6a.png)

I do have a negative values in my CSV though, but I don't think it should present a problem. 

And again, if I set up negative margins like 
![negative2](https://user-images.githubusercontent.com/1167494/35227828-74784f64-ff8f-11e7-872e-47491bf77e95.png)

I get the coordinates, but not the ticks
![negative](https://user-images.githubusercontent.com/1167494/35227846-8015f20e-ff8f-11e7-86af-bdbe3b004bf4.png)

**Comment by BigFatDog** [comment]
I'll setup a Angular 5 + Bootstrap 4 app and try myself. I'll come back to you once I've got a conclusion

**Comment by BigFatDog** [comment]
Hello.
Problem resolved. 0.7.5 has been released with the bug fix. I've verified it in a Angular 5 + Bootstrap 4 application.

The original 2 problems you posted are:
1. nullValueSeparator is not set. this is a bug of vizart-path
2. ticks cannot be seen. vizart-path.css hasn't been imported properly

For problem #1:
1. upgrade vizart-path to version 0.7.5
2. Take a look at the release note for configuration changes

For problem #2
I resolved the problem by registering the vizart-path.css as global stylesheet.
When I import in a component level:
```typescript
@Component({
  selector: 'my-dashboard',
  templateUrl: './dashboard.component.html',
  styleUrls: ['./vizart-tpath.css'],
  styles: [`#my-logout-button { background: #F44336 }`]
})
```
Ng CLI (or some other css processors) transforms vizart-path.css into local css modules. It is the root cause of odd location behaviour.

or: If you want to keep vizart-path.css in a local scope other than global scope (i.e., at component level), I think that should be doable.

By the way, I hooked chart rendering to ngOnInit() , the same as you did.
